src/Vat.sol

```diff
@@ -217,23 +217,23 @@ contract Vat {
         // system is live
         require(live == 1, "Vat/not-live");
 
-        Urn memory urn = urns[i][u];
-        Ilk memory ilk = ilks[i];
+        uint256 rate_ = ilks[i].rate;
         // ilk has been initialised
-        require(ilk.rate != 0, "Vat/ilk-not-init");
+        require(rate_ != 0, "Vat/ilk-not-init");
 
+        Urn memory urn = urns[i][u];
         urn.ink = _add(urn.ink, dink);
         urn.art = _add(urn.art, dart);
-        ilk.Art = _add(ilk.Art, dart);
-
-        int256 dtab = _int256(ilk.rate) * dart;
-        uint256 tab = ilk.rate * urn.art;
-        debt     = _add(debt, dtab);
 
+        uint256 Art_  = _add(ilks[i].Art, dart);
+        int256  dtab  = _int256(rate_) * dart;
+        uint256 debt_ = _add(debt, dtab);
         // either debt has decreased, or debt ceilings are not exceeded
-        require(either(dart <= 0, both(ilk.Art * ilk.rate <= ilk.line, debt <= Line)), "Vat/ceiling-exceeded");
+        require(either(dart <= 0, both(Art_ * rate_ <= ilks[i].line, debt_ <= Line)), "Vat/ceiling-exceeded");
```

Comment:
Could someone explain we don't want short circuiting here? Is it a gas optimization?

Reply:
As I understand it the `either` doesn't want to short circuit as the intention is to revert if `Art_ * rate_` overflows.
The `both` as I understand is just complementary to use the same style (to prevent some of the code will be with `either` and some with `||` / `&&` etc..).

Reply:
Ah okay. Is this necessary though? The checks seem redundant with any reasonable state. If those numbers are overflowing the system is already overall hitting overflow limits.